src/libYARP_dev/src/yarp/dev/IJointCoupling.h

```diff
@@ -0,0 +1,213 @@
+     *
+     * @param[in] physJointPos Physical Joint position
+     * @param[in] physJointTrq Physical Joint torque
+     * @param[out] actAxisTrq  Actuated Axes torque
+     * @return true/false on success/failure
+     */
+    virtual bool convertFromPhysicalJointTrqToActuatedAxisTrq(const yarp::sig::Vector& physJointPos, const yarp::sig::Vector& physJointTrq, yarp::sig::Vector& actAxisTrq) = 0;
+
+    /**
+     * @brief Convert from Actuated Axes to Physical Joint position
+     * This method implements $g(\theta)$.
+     * @param[in] actAxisPos Actuated Axes position
+     * @param[out] physJointPos Physical Joint position
+     * @return true/false on success/failure
+     */
+    virtual bool convertFromActuatedAxisToPhysicalJointPos(const yarp::sig::Vector& actAxisPos, yarp::sig::Vector& physJointPos) = 0;
+
+    /**
+     * @brief Convert from Actuated Axes to Physical Joint velocity
+     * This method implements $g'$ that can be used to compute $\dot{q}(t)$, defined such that:
+     *
+     * $$
+     * \dot{q}(t) = \frac{\partial}{\partial \theta} g(\theta(t)) \dot{\theta}(t) = g'( \theta(t), \dot{\theta}(t))
+     * $$
+     *
+     * @param[in] actAxisPos Actuated Axes position
+     * @param[in] actAxisVel Actuated Axes velocity
+     * @param[out] physJointVel Physical Joint velocity
+     * @return true/false on success/failure
+     */
+    virtual bool convertFromActuatedAxisToPhysicalJointVel(const yarp::sig::Vector& actAxisPos, const yarp::sig::Vector& actAxisVel, yarp::sig::Vector& physJointVel) = 0;
+
+    /**
+     * @brief Convert from Actuated Axes to Physical Joint acceleration
+     *
+     * This method implements $f''$ that can be used to compute $\ddot{\theta}(t)$,  defined such that:
+     *
+     * $$
+     * \ddot{q}(t) = \frac{\partial}{\partial \theta} g'(\theta(t), \dot{\theta}(t)) \dot{\theta}(t) + \frac{\partial}{\partial \dot{\theta}} g'(\theta(t), \dot{\theta}(t)) \ddot{\theta}(t) = g''( \theta(t), \dot{\theta}(t), \ddot{\theta}(t))
+     * $$
+     *
+     * @param[in] actAxisPos Actuated Axes position
+     * @param[in] actAxisVel Actuated Axes velocity
+     * @param[in] actAxisAcc Actuated Axes acceleration
+     * @param[out] physJointAcc Physical Joint acceleration
+     * @return true/false on success/failure
+     */
+    virtual bool convertFromActuatedAxisToPhysicalJointAcc(const yarp::sig::Vector& actAxisPos, const yarp::sig::Vector& actAxisVel, const yarp::sig::Vector& actAxisAcc, yarp::sig::Vector& physJointAcc) = 0;
+
+    /**
+     * @brief Convert from Actuated Axes to Physical Joint acceleration
+     *
+     * @param[in] actAxisPos Actuated Axes position
+     * @param[in] actAxisTrq Actuated Axes torque
+     * @param[out] physJointTrq Physical Joint torque
+     * @return true/false on success/failure
+     */
+    virtual bool convertFromActuatedAxisToPhysicalJointTrq(const yarp::sig::Vector& actAxisPos, const yarp::sig::Vector& actAxisTrq, yarp::sig::Vector& physJointTrq) = 0;
```

Comment:
mmmh the pattern seems broken here:
give the method:
convertFromPhysicalJointPosToActuatedAxisPos
I thought the intention was to have something like:
convertFromActuatedAxis***ToPhysicalJoint***
but now it seems to be:
convertFromActuatedAxisToPhysicalJoint***
Maybe the second one is simpler?

Reply:
The discrepancies you noticed were due to typos, but I am open to accepting the option of using 

```
convertFromActuatedAxisToPhysicalJoint@@@
```
instead of

```
convertFromActuatedAxis@@@ToPhysicalJoint@@@
```

cc @traversaro